core/src/main/java/org/elasticsearch/search/DefaultSearchContext.java

```diff
@@ -290,13 +291,13 @@ public Query buildFilteredQuery(Query query) {
         }
     }
 
-    private static Query createTypeFilter(String[] types) {
+    private Query createTypeFilter(String[] types) {
         if (types != null && types.length >= 1) {
-            BytesRef[] typesBytes = new BytesRef[types.length];
-            for (int i = 0; i < typesBytes.length; i++) {
-                typesBytes[i] = new BytesRef(types[i]);
+            MappedFieldType ft = mapperService().fullName(TypeFieldMapper.NAME);
+            if (ft != null) {
+                // ft might be null if no documents have been indexed yet
```

Comment:
you would have an empty list of types and no mappers at all (including metadata mappers)